dataretrieval/aws.py

```diff
@@ -180,8 +195,15 @@ def get_model_historical_data(start_date, end_date=None):
     rows_with_nan = [idx for idx, is_nan in enumerate(rows_with_nan) if is_nan]
     df.drop(axis=0, index=rows_with_nan, inplace=True)
 
-    # TODO Remove rows with outlier data
+    # Go through all the time points and delete outliers
+    for time_point in df:
+        for attr in df.columns:
+            if ATTR_BOUNDARIES[attr][MIN] <= time_point[attr] <= ATTR_BOUNDARIES[attr][MAX]:
+                continue
+            # Out of boundaries
+            df.drop(time_point)
```

Comment:
I think you need `df.drop(time_point, inplace=True)` here otherwise `drop` will return a copy.